### PR TITLE
add "including one of" segment for multiselect fields

### PR DESCRIPTION
**Pull request by kuzmany**
**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | 
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
This PR added support for **include one of** expression for multi select field.
By default include/exclude expect selected values in condition.
**Include one of** expect at least on of selected value.

For example: 
- include - if we expect fruit in filter - orange, lemon - contact must these values in profile
- **Include one of**- If we expect fruit in filter - orange, lemon - contact could have at least one of them in profile

image

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com). On  your own test enviroment require assets generate **php app/console m:a:g**
2. Create multiselect contact field  with multiple options
3. Create segment with filter **Include one of** some values  of that multiselect field
4. Create contact with one of that values
5. Segment rebuild
6. Your contact should build in your segment
7. If you change expression from **Include one of** to include, you contact should not be passed by segment filter  and should remove from it

### CODE REVIEW

**Comment by npracht** [comment]
Hi there!
It has been decided to not create any extra Mautic 2 minor versions (which means no more features in M2) knowing that Mautic 3 is coming very soon.

**We now want to integrate your contribution in the Mautic 3 roadmap as 3.1.0 candidate.**

## How to do?
1. Check if your feature or enhancement is still missing and relevant in Mautic 3
2. **If Yes**: rebase your PR against the 3.x branch
3. **If No**: please consider closing your PR if you are absolutely sure that the feature has made it into Mautic 3 already

**Please report results by commenting on your PR to make us administration easier.**
______
You can more information on how to do all of that on [this blog post "Getting you PR ready for Mautic 3"](https://www.mautic.org/blog/developer/getting-your-pr-ready-mautic-3).

**Comment by kuzmany** [comment]
Rebased

**Comment by RCheesley** [comment]
@cla-bot check

**Comment by RCheesley** [comment]
@kuzmany great feature - I've tagged it for the 3.2 release.

File: app/bundles/LeadBundle/Translations/en_US/messages.ini [hunk 1]
```diff
@@ -429,6 +429,7 @@ mautic.lead.list.form.operator.equals="equals"
 mautic.lead.list.form.operator.greaterthan="greater than"
 mautic.lead.list.form.operator.greaterthanequals="greater than or equal"
 mautic.lead.list.form.operator.in="including"
+mautic.lead.list.form.operator.inX="including one of"
```
Comment (wise-gorilla) [inline app/bundles/LeadBundle/Translations/en_US/messages.ini]: including any of

**Review by npracht — APPROVED** [review]
👍

**Comment by npracht** [comment]
@Wise-Gorilla @RCheesley any of you can confirm that works as expected ? :)

**Comment by RCheesley** [comment]
@kuzmany seems that we need to [improve the test coverage ](https://github.com/mautic/mautic/pull/8169#issuecomment-719470450)on this one!

**Comment by kuzmany** [comment]
This need be re-done with subscribers once If segments refactoring will be merge https://github.com/mautic/mautic/pull/8649
Then I close it now

**Pull request closed without merging**